pkg/apis/metrics/types.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api"
+)
+
+// +genclient=true
+// +resourceName=nodes
+// +readonly=true
```

Comment:
Don't you need a "nonNamespaced" tag like here: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L2862 ?

Also, care to open an issue for us to add an example of this to the sample-apiserver?